crates/astria-conductor/src/conductor.rs

```diff
@@ -244,13 +400,33 @@ impl Conductor {
             info!("all tasks shut down regularly");
         }
         info!("shutting down");
+
+        if should_restart {
+            RestartToken::Restart
+        } else {
+            RestartToken::Shutdown
+        }
     }
 }
 
 #[instrument(skip_all)]
-fn report_exit(exit_reason: eyre::Result<&str>, message: &str) {
+fn report_exit(exit_reason: &eyre::Result<&str>, message: &str) {
     match exit_reason {
         Ok(reason) => info!(%reason, message),
         Err(reason) => error!(%reason, message),
     }
 }
+
+#[instrument(skip_all)]
+fn check_for_restart(err: &eyre::ErrReport) -> bool {
```

Comment:
Can you create a a simple unit test with a tonic error somewhere in the source chain?

Reply:
There's currently a blackbox test in `soft_and_firm()` to make sure the conductor restarts. Do you want to implement a unit test in addition to this? I originally had purposefully not done this because I remember you had said not to add any new unit tests to conductor for the chain ID checks, but happy to add one if needed.

Reply:
Yeah, not a unit test for the restart logic of the full task (which the blackbox test covers) - but walking through a chain of errors is a good unit test.